sendgrid.go

```diff
@@ -47,3 +55,36 @@ func NewSendClient(key string) *Client {
 	request.Method = "POST"
 	return &Client{request}
 }
+
+// extractEndpoint extracts the endpoint from a baseURL
+func extractEndpoint(link string) (string, error) {
```

Comment:
Can we make it private?

Reply:
It is private only. In Go, the functions having the first letter of their name as capital are public and others are private. So here it is private only